stdlib/Serialization/src/Serialization.jl

```diff
@@ -1522,4 +1522,30 @@ function deserialize(s::AbstractSerializer, ::Type{Base.StackTraces.StackFrame})
     return Base.StackTraces.StackFrame(func, file, line, nothing, from_c, inlined, pointer)
 end
 
+function serialize(s::AbstractSerializer, lock::Base.AbstractLock)
+    # assert_havelock(lock)
+    serialize_cycle_header(s, lock)
+    nothing
+end
+
+function deserialize(s::AbstractSerializer, ::Type{T}) where T<:Base.AbstractLock
+    lock = T()
```

Comment:
This assumes that all fields of an `AbstractLock` are lock states, right? I think it's conceivable to have locks with parameters (e.g., back-off delays). Maybe we can serialize only `const` fields? It's somewhat too magical but kinda cool too.

In a long run, it'd be nice to have a `Serialization` API to declare the part of the value that shouldn't be serialized. I needed a rather ugly hack to avoid serialization of process-local scratch space: 
https://github.com/JuliaFolds/FLoops.jl/blob/255382d84c187ec10521c0166412e5dfb213471b/src/scratchspace.jl#L45-L48 I don't think we need to solve it here but `AbstractLock` sounds like a good case study.

Reply:
Talked to Jeff, and we discussed how this is better than the current situation, and if someone needs something fancier, they can still implement that.

Reply:
Yes, I agree this PR is an improvement overall.